projectional/src/main/java/jetbrains/jetpad/projectional/cell/BaseProjectionalSynchronizer.java

```diff
@@ -649,4 +724,28 @@ public int size() {
       return myRoleSynchronizer.getMappers().size();
     }
   }
+
+  private class ContentToItemMapper<ContentT> {
```

Comment:
The name of the class is confusing. You are saving both fromContent and toContent but the name of class reflects only one direction.